### PR TITLE
Added Castling and fixed piece names

### PR DESCRIPTION
**Pull request by benkeun**

### CODE REVIEW

File: v1/js/c.js [hunk 1]
```diff
@@ -81,9 +81,18 @@ var c = {
...
+            console.log(destPiece.id);
+            if (destPiece.id == 'rook' &&   !(chess.getPieceAt(x1,y1).hasMoved() || destPiece.hasMoved())){
+              if ((y1< y2 && chess.isClearPath(x1,y1,x2,y2-1))||(y1> y2 && chess.isClearPath(x1,y1,x2,y2+1))) {
+              return true;
+              
+              }
+            }else{
+              return false;
+            }
           }
```
Comment (joshgrift) [inline v1/js/c.js]: I'm assuming this relates to castling?

File: v1/js/c.js [hunk 2]
```diff
@@ -81,9 +81,18 @@ var c = {
         b:[[7,3]],
       },
       canMove:function(x1,y1,x2,y2){
-        if(chess.getPieceAt(x2,y2)){
-          if(chess.getPieceAt(x2,y2).team == this.team){
-            return false;
+        var destPiece = chess.getPieceAt(x2,y2);
+        if(destPiece){
+          if(destPiece.team == this.team){
+            console.log(destPiece.id);
```
Comment (joshgrift) [inline v1/js/c.js]:
```suggestion       
```

File: v1/js/chess.js [hunk 3]
```diff
@@ -161,21 +161,35 @@ var Chess = function(element,c,onComplete){
...
           this.turn = 'w';
         }
-
         // disable the mvoe so thr player can't have fun with the apponent
         this.moveDisabled = true;
+        if (currentPiece &&this.dragging.team==currentPiece.team){
+          if (this.dragging.y < currentPiece.y){
+            this.dragging.move(Math.floor(x/c.t),Math.floor(y/c.t-2));
+            currentPiece.move(Math.floor(x/c.t),Math.floor(y/c.t-3));
+            console.log("not this side");
```
Comment (joshgrift) [inline v1/js/chess.js]:
```suggestion
```

File: v1/js/c.js (hunk 1 above)
Comment (benkeun) [reply v1/js/c.js]: Castling is the act of moving the king and one of the rooks from the corner to the middle as a way of defending the King. So if they drag their king to one of the rooks which is still in the corner and the path is clear they will both move to be swapped on the tiles between them.

**Review by joshgrift — APPROVED** [review]

**Pull request merged**